utils/terrain/src/map.rs

```diff
@@ -0,0 +1,120 @@
+use std::mem;
+use std::ops::{Index, IndexMut};
+
+/// A simple "flat" map based on a sorted vector
+///
+/// See the [module level documentation][super] for why one should use this.
+///
+/// The API is roughly similar to that of [`std::collections::HashMap`], though it
+/// requires `Ord` instead of `Hash`.
+#[derive(Clone, Debug)]
+pub struct VecMap<K, V> {
+    values: Vec<(K, V)>
+}
+
+impl<K, V> VecMap<K, V> {
+    /// Construct a new [`VecMap`]
+    pub fn new() -> Self {
+        VecMap::default()
+    }
+
+    /// The number of elements in the [`VecMap`]
+    pub fn len(&self) -> usize {
+        self.values.len()
+    }
+
+    /// Remove all elements from the [`VecMap`]
+    pub fn clear(&self) -> usize {
+        self.values.clear()
+    }
+
+    /// Reserve capacity for `additional` more elements to be inserted into
+    /// the [`VecMap`] to avoid frequent reallocations.
+    ///
+    /// See [`Vec::reserve()`] for more information.
+    pub fn reserve(&self, additional) -> usize {
+        self.values.reserve(additional)
+    }
+}
+
+impl<K: Ord, V> VecMap<K, V> {
+    /// Get the value associated with `key`, if it exists.
+    pub fn get(&self, key: &K) -> Option<&V> {
```

Comment:
Since there are 2 lifetimes here, do you need to make them explicit?  The lifetime of `&self` is the same as `&V`, but the lifetime of `&K` is shorter.

Reply:
Nah, elision will match `&V` to `&self`. Anything else would not have compiled anyway